src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,245 @@
+package storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import exceptions.BrockException;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.ToDo;
+
+/**
+ * Class to create, and interact with save file.
+ */
+public class Storage {
+    private static final String FILE_PATH = "./src/main/java/saveFile.txt";
+    private static final File SAVE_FILE = new File(FILE_PATH);
+
+    /**
+     * Removes closing bracket from the body string of each task.
+     * So that the correct date or time can be extracted for deadline and event tasks.
+     *
+     * @param target String fragment with closing bracket to be removed.
+     * @return String without the closing bracket.
+     */
+    private String removeCloseBracket(String target) {
```

Comment:
Would it be productive to have a check here to verify that the last character in target is indeed a close bracket, just to account for possible cases of corrupted data?

Reply:
Agreed, good idea to include checks for save file corruption in general throughout the code 💯